aptos-move/block-executor/src/executor.rs

```diff
@@ -709,8 +705,49 @@ where
                     accumulated_fee_statement.add_fee_statement(&fee_statement);
                     counters::update_sequential_txn_gas_counters(&fee_statement);
 
+                    // TODO for materialization, we need to understand what we read,
+                    // or do exchange on every transaction, so this logic might change
+                    let mut second_phase = Vec::new();
+                    let mut updates = HashMap::new();
+                    for (id, change) in output.aggregator_v2_change_set().into_iter() {
+                        match change {
+                            AggregatorChange::Create(value) => {
+                                updates.insert(id, value);
+                            },
+                            AggregatorChange::Apply(apply) => {
+                                match apply.get_apply_base_id(&id) {
+                                    ApplyBase::Previous(base_id) => {
+                                        updates.insert(
+                                            id,
+                                            apply
+                                                .apply_to_base(
+                                                    data_map.fetch_aggregator(&base_id).unwrap(),
+                                                )
+                                                .unwrap(),
```

Comment:
What if apply fails?

Reply:
there is nothing speculative here, so should never fail.

and there is nothing really we can fallback from sequential execution

Reply:
I don't understand. If there is delta application failure, you call unwrap? We should abort the transaction here no?

Reply:
discussed offline, there is no delta application failure that bubbles up here in sequential execution